docs/user/modeling.md

```diff
@@ -0,0 +1,144 @@
+|---------------------------|---------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------|
+| Normal fields             | Has to match ORM exactly  | Make sure that the name matches the name in the ORM model.                                                                                       |
+| Custom fields             | Field name doesn't matter | Use `nautobot_ssot.contrib.CustomFieldAnnotation`                                                                                                |
+| Many-to-one relationships | Django lookup syntax      | See [here](https://docs.djangoproject.com/en/3.2/topics/db/queries/#lookups-that-span-relationships) - your model fields need to use this syntax |
+| One-to-many relationships | Has to match ORM exactly  | In case of a generic foreign key see [here](#special-case-generic-foreign-key)                                                                   | 
+
+
+## Normal Fields
+
+For normal, non-relationship, non-custom fields on a model, all you need to do is to ensure that field name on your SSoT model class matches that of the Nautobot model class. To ensure this is the case, you can check out the model either by reading the corresponding source code on [GitHub](https://github.com/nautobot/nautobot) or through the [Nautobot Shell](https://docs.nautobot.com/projects/core/en/stable/administration/nautobot-shell/) - in there you can use commands like `dir(Tenant)` to get an overview of available fields for the different models.
+
+!!! note
+    If you want to sync data into a model provided by a [Nautobot App](https://docs.nautobot.com/projects/core/en/stable/plugins/), you need to navigate to its respective source code repository. Furthermore, you may need to manually import the model if you're using the Nautobot Shell approach.
+
+## Custom Fields
+
+For [custom fields](https://docs.nautobot.com/projects/core/en/stable/models/extras/customfield/), you will need to use the `nautobot_ssot.contrib.CustomFieldAnnotation` class. Given a custom field called "Test" on the circuit provider model, this is how you could map that custom field into the diffsync model field `test_custom_field`:
+
+```python
+from typing_extensions import Annotated  # Python<3.9 - you need to install this module
+# from typing import Annotated  # Python>=3.9
+from nautobot.circuits.models import Provider
+from nautobot_ssot.contrib import NautobotModel, CustomFieldAnnotation
+
+class ProviderModel(NautobotModel):
+    _model = Provider
+    _modelname = "provider"
+    _identifiers = ("name",)
+    _attributes = ("custom_field",)
+
+    name: str
+    test_custom_field: Annotated[str, CustomFieldAnnotation(name="Test")]
+```
+
+!!! note
+    Defining a `CustomFieldAnnotation` variable is necessary since custom field names may include spaces, which are un-representable in Python object field names.
```

Comment:
why not use a name with an space in the example?

Reply:
Fixed your point @chadell, and good point @glennmatthews - I have added a TODO comment for 2.0